Sanitizer.cmake

```diff
@@ -0,0 +1,50 @@
+# Used to turn on certain sanitizers, if available:
+# 
+# clang:
+#    -dumpcompiler: 4.2.1
+#       address
+#       thread
+#       undefined
+# gcc:
+#    -dumpcompiler: 4.8.3
+#       address
+#       thread
+#
+# Note: This modifies CMAKE_CXX_FLAGS and CMAKE_EXE_LINKER_FLAGS
+# Usage: add this when doing cmake configuration: -DSANITIZER=address
+
+option(SANITIZER "Sanitizer: one of address/thread/undefined, depending on compiler" OFF)
```

Comment:
I was going to suggest adding back the SANITIZE option as @eile proposed, but since the gcc documentation states "there is no -fsanitize=all, as some sanitizers cannot be used together" I'm sure if this makes sense.